chia/wallet/did_wallet/did_wallet.py

```diff
@@ -189,18 +193,18 @@ def type(cls) -> uint8:
     def id(self):
         return self.wallet_info.id
 
-    async def get_confirmed_balance(self, record_list=None) -> uint64:
+    async def get_confirmed_balance(self, record_list=None) -> uint128:
         if record_list is None:
             record_list = await self.wallet_state_manager.coin_store.get_unspent_coins_for_wallet(self.id())
 
-        amount: uint64 = uint64(0)
+        amount: uint128 = uint128(0)
         for record in record_list:
             parent = self.get_parent_for_coin(record.coin)
             if parent is not None:
-                amount = uint64(amount + record.coin.amount)
+                amount = uint128(amount + record.coin.amount)
 
         self.log.info(f"Confirmed balance for did wallet is {amount}")
-        return uint64(amount)
+        return uint128(amount)
```

Comment:
nit: I'd probably drop the uint128(), but I understand if you'd prefer to keep it for clarity.

Reply:
Thanks, I was considering dropping it, but kept it for clarity

Reply:
I believe that this is not about clarity but rather correctness.  Our fixed width integer types need some work.  1) they are really expensive to create since they serialized and deserialize on creation and 2) they don't really retain their types very well.

```pycon
>>> from chia.util.ints import uint128
>>> x = uint128(0)
>>> y = x + x
>>> type(y)
<class 'int'>
>>> x += uint128(1)
>>> type(x)
<class 'int'>
```